beacon/engine/types.go

```diff
@@ -275,7 +275,7 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 	// ExecutableData before withdrawals are enabled by marshaling
 	// Withdrawals as the json null value.
 	var withdrawalsRoot *common.Hash
-	if data.WithdrawalsRoot != nil {
+	if config.IsOptimismIsthmus(data.Timestamp) && data.WithdrawalsRoot != nil {
```

Comment:
I am surprised that this doesn't break any tests as when op stack is disabled, this branch will never be hit

Reply:
The provided withdrawals-root is fine to ignore when the same root is re-computed from the actual withdrawals.
That said, yes this `IsOptimismIsthmus(data.Timestamp) &&` shouldn't be here. Instead, I think we should have a separate `if IsOptimismIsthmus`, that then checks if the withdrawals-root is set, and error otherwise. And leave the existing withdrawals-root check the same, since it just prevents recomputing it if it's set already.

Reply:
@protolambda Your PR [requires](https://github.com/ethereum-optimism/op-geth/pull/383/files#diff-50e0dca0bd0ac667ca6a4a3b762c9346030f346fdd9b48f3f83be675a43423eaR278-R283) that the withdrawals list be empty and returns an error otherwise.

To @tynes 's point, as-is, it breaks a bunch of existing tests [here](https://github.com/ethereum-optimism/op-geth/blob/optimism/eth/catalyst/api_test.go#L1341-L1350) that explicitly set a non-zero withdrawals list.

I was thinking that the check you added in your PR should only be enforced if Isthmus is active (which obviously implies that op-stack is enabled). Am I missing something?

Reply:
That's an error in my PR draft, we should change that. You're right that there needs to be an optimism fork check on it. But the one like implemented here is broken as described above.